src/services/user.js

```diff
@@ -64,13 +71,62 @@ export const loginUser = async (req) => {
 
   const loginUser = await User.findById(user._id).select("-password");
 
+  const payload = { userid: loginUser._id, role: loginUser.role };
+  const key = process.env.ACCESS_TOKEN_SECRET;
+  const expiresIn = process.env.ACCESS_TOKEN_EXPIRY;
+
+  const jwtToken = createToken(payload, key, expiresIn);
+
   const options = {
     httpOnly: true,
     secure: true,
   };
 
   return {
     options,
-    loginUser
+    loginUser,
+    jwtToken
   };
-};
\ No newline at end of file
+};
+
+export const getUserDetails = async (req) => {
+
+  const { userid } = req.user;
+
+  if (!userid) {
+    throw new CustomError(
+      statusCodes?.notFound,
+      Message?.notFound,
+      errorCodes?.not_found,
+    );
+  }
+
+  const userData = await User.findById(userid).select("-password");
```

Comment:
can we add sort and lean mentod of mongoose

Reply:
sir, i don't understand what changes we need